CleverPush/CleverPush/Source/CPAppBannerController.m

```diff
@@ -334,22 +379,25 @@ - (void)userContentController:(WKUserContentController*)userContentController
 }
 
 - (void)webView:(WKWebView *)webView didFinishNavigation:(WKNavigation *)navigation {
-    
     [webView evaluateJavaScript:@"document.readyState" completionHandler:^(id _Nullable result, NSError * _Nullable error) {
         [webView evaluateJavaScript:@"document.body.scrollHeight" completionHandler:^(id _Nullable result, NSError * _Nullable error) {
             CGFloat height = [result floatValue];
-            if (height > UIScreen.mainScreen.bounds.size.height) {
-                self.bannerBody.frame = CGRectMake(20, 70 , self.bannerBody.frame.size.width, UIScreen.mainScreen.bounds.size.height - 140);
-                self.bannerBodyContent.frame = CGRectMake(20, 20, self.bannerBodyContent.frame.size.width, self.bannerBody.frame.size.height - 40);
-                webView.frame = CGRectMake(0, 0, self.bannerBodyContent.frame.size.width, self.bannerBodyContent.frame.size.height);
-            }
-            else
-            {
-                CGFloat ScreenHeight = UIScreen.mainScreen.bounds.size.height;
-                CGFloat popupY = (ScreenHeight - height) / 2;
-                self.bannerBody.frame = CGRectMake(20, popupY , self.bannerBody.frame.size.width, height + 40);
-                self.bannerBodyContent.frame = CGRectMake(20, 20, self.bannerBodyContent.frame.size.width, self.bannerBody.frame.size.height - 40);
-                webView.frame = CGRectMake(0, 0, self.bannerBodyContent.frame.size.width, self.bannerBodyContent.frame.size.height);
+            if (webView == self.webBanner) {
+                if (height > UIScreen.mainScreen.bounds.size.height) {
+                    self.bannerBody.frame = CGRectMake(20, 70 , self.bannerBody.frame.size.width, UIScreen.mainScreen.bounds.size.height - 140);
+                    self.bannerBodyContent.frame = CGRectMake(20, 20, self.bannerBodyContent.frame.size.width, self.bannerBody.frame.size.height - 40);
+                    webView.frame = CGRectMake(0, 0, self.bannerBodyContent.frame.size.width, self.bannerBodyContent.frame.size.height);
+                }
+                else
+                {
+                    CGFloat ScreenHeight = UIScreen.mainScreen.bounds.size.height;
+                    CGFloat popupY = (ScreenHeight - height) / 2;
+                    self.bannerBody.frame = CGRectMake(20, popupY , self.bannerBody.frame.size.width, height + 40);
+                    self.bannerBodyContent.frame = CGRectMake(20, 20, self.bannerBodyContent.frame.size.width, self.bannerBody.frame.size.height - 40);
+                    webView.frame = CGRectMake(0, 0, self.bannerBodyContent.frame.size.width, self.bannerBodyContent.frame.size.height);
+                }
+            }else{
```

Comment:
empty else block?

Reply:
actually there are two WKWebView in the same controller file to bifurcate the type of WKWebView on the delegate method it was developed before we havent decided about the height about the HTML block so it was forgotten to remove